src/cascade/executor/dismod_runner.py

```diff
@@ -134,6 +127,22 @@ def reduce_process_priority():
     os.nice(19)
 
 
+@asyncio.coroutine
+def _read_pipe(pipe, result, callback=lambda text: None):
+    """Read from a pipe until it closes.
+
+    Args:
+        pipe: The pipe to read from
+        result: a list to accumulate the output into
+        callback: a callable which will be invoked each time data is read from the pipe
+    """
+    while not pipe.at_eof():
+        text = yield from pipe.read(2 ** 16)
+        text = text.decode("utf-8")
+        result.append(text)
```

Comment:
This is a choice to keep the whole stdout in memory and to send everything-so-far to the callback.
The goal of this callback, currently, is to append to a text file that can be read, and refreshed, through EpiViz and a browser. It might work better to send up to the last line, but that wouldn't be decided here, I suppose.

Reply:
I'm preserving the existing behavior. Rather than having the mutable accumulator and the callback, we could just have the callback which sends to the logs and/or does something else. I don't fully understand what's happening with this code so I wanted to keep my changes local.